Reely/Controller/FilmViewController.swift

```diff
@@ -0,0 +1,248 @@
+        alertController.view.tintColor = plum
+        present(alertController, animated: true, completion: nil)
+    }
+    
+    
+    /* method to initiate the API Request  */
+    func getData(title : String){
+        
+        FilmManager.shared.getFilmData(query: title, page: String (page))
+        { [weak self ] results in
+            guard let self = self else {return}
+            switch results {
+                case .success(let results):
+                    
+                    self.filmArray = results
+                    self.search = title
+                    self.updateSearchSuggestions(title: title)
+                    self.filmTableView.reloadData()
+                
+                case .failure(let error):
+                    DispatchQueue.main.async {
+                        self.error(error: error)
+                        
+                }
+                
+            }
+        }
+        
+    }
+    
+    /* call this in the viewDidLoad and setup the child VC  */
+    func setupView(){
+        updateView()
+        reloadChildViews()
+    }
+    
+    /* reload the tableView of the childVC  */
+    func reloadChildViews(){
+        suggestionsController?.searchTableView.reloadData()
+    }
+    
+    /* this is where showing the controller happens  */
+    // MARK: - updateView Method
+    private func updateView() {
+        
+        if searchBar?.searchTextField.isEditing ?? false {
+            presentSearchController(resultsSearchController)
+            
+            self.reloadChildViews()
+            
+        }
+        
+
+        /* this method is to add a childView to the parentView  */
+        // MARK: - add a child View
+        func add(asChildViewController viewController: UIViewController) {
+            // Add Child View as Subview
+            view.addSubview(viewController.view)
+            
+            // Configure Child View
+            viewController.view.frame = view.bounds
+            viewController.view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        }
+        
+        
+    }
+    
+    /* this method is to remove a childView from the superView but I didn't end up needing it.   */
+    // MARK: - remove a child view
+    func remove(asChildViewController viewController: UIViewController) {
+        // Notify Child View Controller
+        viewController.willMove(toParent: nil)
+        
+        // Remove Child View From Superview
+        viewController.view.removeFromSuperview()
+        
+        // Notify Child View Controller
+        viewController.removeFromParent()
+    }
+}
+/* this is for setting the tint color to the same as the one in the storyboard.
+ I decided that this was easier than finding the rgb values*/
+// MARK: - hexStringToUIColor
+func hexStringToUIColor (hex:String) -> UIColor {
+    var cString:String = hex.trimmingCharacters(in: .whitespacesAndNewlines).uppercased()
+    
+    if (cString.hasPrefix("#")) {
```

Comment:
in general, avoid using parens when they're not needed, as in this case `if cString.hasPrefix("#")` will suffice